### PR TITLE
Styling Guide Creation

### PR DESCRIPTION
**Pull request by warpling**
Alright, still a work in progress with some problems, but it's a step!

# What Happened
* **Guide creation got styled**
 * New guide creation nav (with checks 'n' stuff)
 * All around better responsiveness
 * More consistent styling/typography
 * Flatter button/selector style
 * Cool plant selection thingy (Try It Out!™)
 * New Life Stage selection page
 * Tweaked action selection style on step 3
 * General refactoring

* **New and old bugs** (Need your help here @RickCarlino and @simonv3!)
 * Crop selection cancelation
   * I introduced some new directives so that when the crop selection is cleared focus would come back to the search field and the input would be cleared. It isn't working. (Autofocus on the input when the page loads is though)
 * Step 3 has a *sacrificial form element*. For reasons unknown, Angular stuff is consuming the first form element. I fed it an empty one so it wouldn't touch the real one (`<form name="stagesForm">`). I'm not superstitious. We need to figure this out and remove it.
 * We need to double-check that everything in guide creation is actually saving.
 * "Other" pop-overs on Step 3 are broke

Sorry this is a crappy PR!

This is a diagram of how I felt today when I'd fix one problem and create another.
2qy0kje

cc @simonv3 @RickCarlino @roryaronson @growdigital

### CODE REVIEW

File: app/assets/javascripts/guides/new.js [hunk 1]
```diff
@@ -48,6 +48,49 @@ openFarmApp.directive('formChecker', function(){
...
+      scope.$on('clearOn', function(e, name) {
+        if(name === attr.clearOn) {
+          elem[0].value = "";
+        }
+      });
+   };
+});
+
+// TODO: Does this belong here?
+// Source: http://stackoverflow.com/questions/14833326/how-to-set-focus-on-input-field/14837021#14837021
```
Comment (simonv3) [inline app/assets/javascripts/guides/new.js]: Eh

File: app/assets/javascripts/angular-libs/angular.openfarm.js [hunk 2]
```diff
@@ -94,6 +94,7 @@ openFarmModule.directive('multiRowSelect', [
       controller: ['$scope', '$element', '$attrs',
         function ($scope, $element, $attrs) {
           $scope.multiSelectType = $attrs.multiSelectType || 'checkbox';
+          $scope.multiSelectOverflowCount = $attrs.multiSelectOverflowCount || 3;
```
Comment (simonv3) [inline app/assets/javascripts/angular-libs/angular.openfarm.js]: I'll fix this when I push this fixed directive.

File: app/assets/stylesheets/main.css.scss [hunk 3]
```diff
@@ -42,9 +37,8 @@ $compatibility-low-color-bg: rgba(178, 58, 44, 0.2);
 $compatibility-med-color-bg: rgba(251, 173, 47, 0.2);
 $compatibility-high-color-bg: rgba(132, 201, 63, 0.2);
 
-@import 'styles/presets';
-
 @import 'styles/foundation_and_overrides';
+@import 'styles/presets';
```
Comment (simonv3) [inline app/assets/stylesheets/main.css.scss]: Can you pull in the most recent master? @growdigital moved the variables in it.

File: app/assets/stylesheets/styles/_presets.css.scss [hunk 4]
```diff
@@ -28,9 +28,14 @@
   display: inline-block;
 }
 
-@mixin button-box-shadow(){
-  -moz-box-shadow: 0px 2px 0px 0px rgba(0,0,0,0.08), 0px 4px 10px 0px rgba(0,0,0,0.10), inset 0px 1px 0px 1px rgba(255,255,255,0.1);
-  box-shadow: 0px 2px 0px 0px rgba(0,0,0,0.08), 0px 4px 10px 0px rgba(0,0,0,0.10), inset 0px 1px 0px 1px rgba(255,255,255,0.1);
+@mixin manual-font-awesome($glyph) {
```
Comment (simonv3) [inline app/assets/stylesheets/styles/_presets.css.scss]: oh?

File: app/assets/stylesheets/styles/guides/_new_guides.css.scss [hunk 5]
```diff
@@ -1,5 +1,4 @@
-// You can't nest keyframes
-
+// TODO: This should not be here. This makes me unhappy.
 @-webkit-keyframes fa-spin-image-load {
```
Comment (simonv3) [inline app/assets/stylesheets/styles/guides/_new_guides.css.scss]: I couldn't figure out what to do with this. Maybe we just need our own keyframe and not use `fa-spin` in this case?

File: app/assets/stylesheets/styles/guides/_new_guides.css.scss [hunk 6]
```diff
@@ -43,111 +134,137 @@
...
     }
 
-    .preview{
+    .preview {
       border-radius: $global-radius;
       opacity: .5;
     }
 
-    .clickable{
+    .clickable {
```
Comment (simonv3) [inline app/assets/stylesheets/styles/guides/_new_guides.css.scss]: I'm hoping you just did find & replace on this, though I'm totally picturing you going through these one by one and fixing them.

File: app/assets/javascripts/guides/new.js (hunk 1 above)
Comment (warpling) [inline app/assets/javascripts/guides/new.js]: Bane of my existence

File: app/assets/javascripts/guides/new.js [hunk 7]
```diff
@@ -48,6 +48,49 @@ openFarmApp.directive('formChecker', function(){
   };
 });
 
+openFarmApp.directive('clearOn', function() {
+   return function(scope, elem, attr) {
+      scope.$on('clearOn', function(e, name) {
+        if(name === attr.clearOn) {
+          elem[0].value = "";
```
Comment (warpling) [inline app/assets/javascripts/guides/new.js]: Hmm in my mind, one of these is a symbol and the other is a string, but maybe that's just me ¬_¬

File: app/assets/stylesheets/main.css.scss (hunk 3 above)
Comment (warpling) [inline app/assets/stylesheets/main.css.scss]: Was it very recent? I pulled in master about two hours ago?

File: app/assets/stylesheets/styles/_presets.css.scss (hunk 4 above)
Comment (warpling) [inline app/assets/stylesheets/styles/_presets.css.scss]: Is this okay? I started moving things towards a feel more similar to the Treehouse and in the end  reduced the shadows to the point where the mixin didn't make sense. How do you think the buttons look?

File: app/assets/stylesheets/styles/guides/_new_guides.css.scss (hunk 5 above)
Comment (warpling) [inline app/assets/stylesheets/styles/guides/_new_guides.css.scss]: Yea it was weird I tried moving this to `main` but it didn't seem to be detected in `_new_guides`…

File: app/assets/stylesheets/styles/guides/_new_guides.css.scss (hunk 6 above)
Comment (warpling) [inline app/assets/stylesheets/styles/guides/_new_guides.css.scss]: Hah I wrote a regex ;)

File: app/assets/stylesheets/main.css.scss (hunk 3 above)
Comment (simonv3) [inline app/assets/stylesheets/main.css.scss]: Yeah I did it about two hours ago. (this should probably fix the merge conflict)

File: app/assets/stylesheets/styles/_presets.css.scss (hunk 4 above)
Comment (simonv3) [inline app/assets/stylesheets/styles/_presets.css.scss]: The oh was for the manual-font-awesome mixin, just wondering what it does and why?

File: app/assets/javascripts/guides/new.js (hunk 1 above)
Comment (warpling) [inline app/assets/javascripts/guides/new.js]: Got it, hound.

**Pull request merged**